### DISCUSSION TITLE
Error consulta datos

### DISCUSSION THREAD

**Discussion by cdv13** [santiemanuel/milProgramadoresQA#31] [Preguntas y Respuestas] [answered]
Hola Santi, a que se puede deber este error:

`near "@gmail": syntax error`

quiero usar unos datos de la tabla usuario y me doy con este error.

la consulta es la siguiente:
`tipo=conexion.read(f"SELECT tipo FROM usuario WHERE nombre_US={self.__nombreUsuario} and clave={self.__clave}")`

**Comment by santiemanuel** [comment, marked as answer]
Hola Clara, eso pasa porque en tu f-string al hacer eso te faltaría las comillas para la parte de string de la consulta sql. Te recomiendo separarlo como sigue:

```py
consulta = "SELECT tipo FROM usuario WHERE nombre_US=? and clave=?"

# Bind the values to the placeholders in the query
tipo = conexion.read(consulta, (self.__nombreUsuario, self.__clave))
```

Acá ya no te preocupas por las comillas, pero vas a tener que agregar un parámetro a tu método read. Si querés mantener tu versión original (pero menos segura en términos de sql) ponés:
`tipo=conexion.read(f"SELECT tipo FROM usuario WHERE nombre_US='{self.__nombreUsuario}' and clave='{self.__clave}'")`

**Reply by cdv13** [reply]
Yo estaba probando con la segunda forma, y me saltaba : "no such table: usuario". Ahora voy a ver la primera

**Reply by cdv13** [reply]
Ahora probé la primera, y tb me sale :"no such table: usuario"; y si imprimo la consulta con ambos métodos me da "None"
